sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClientBuilder.java

```diff
@@ -100,98 +69,32 @@ public BlobClient buildBlobClient() {
-     * only use this when the blob is known to be an append blob.
-     *
-     * @return a {@link AppendBlobClient} created from the configurations in this builder.
-     * @throws NullPointerException If {@code endpoint}, {@code containerName}, or {@code blobName} is {@code null}.
-     */
-    public AppendBlobClient buildAppendBlobClient() {
-        return new AppendBlobClient(buildAppendBlobAsyncClient());
-    }
-
-    /**
-     * Creates a {@link AppendBlobAsyncClient} based on options set in the Builder. AppendBlobAsyncClients are used to
-     * perform append blob specific operations such as {@link AppendBlobAsyncClient#appendBlock(Flux, long) append
-     * blob}, only use this when the blob is known to be an append blob.
-     *
-     * @return a {@link AppendBlobAsyncClient} created from the configurations in this builder.
-     * @throws NullPointerException If {@code endpoint}, {@code containerName}, or {@code blobName} is {@code null}.
-     */
-    public AppendBlobAsyncClient buildAppendBlobAsyncClient() {
-        return new AppendBlobAsyncClient(constructImpl(), snapshot, cpk);
-    }
-
-    /**
-     * Creates a {@link BlockBlobClient} based on options set in the Builder. BlockBlobClients are used to perform
-     * generic upload operations such as {@link BlockBlobClient#uploadFromFile(String) upload from file} and block blob
-     * specific operations such as {@link BlockBlobClient#stageBlock(String, InputStream, long) stage block} and {@link
-     * BlockBlobClient#commitBlockList(List)}, only use this when the blob is known to be a block blob.
-     *
-     * @return a {@link BlockBlobClient} created from the configurations in this builder.
-     * @throws NullPointerException If {@code endpoint}, {@code containerName}, or {@code blobName} is {@code null}.
-     */
-    public BlockBlobClient buildBlockBlobClient() {
-        return new BlockBlobClient(buildBlockBlobAsyncClient());
-    }
-
-    /**
-     * Creates a {@link BlockBlobAsyncClient} based on options set in the Builder. BlockBlobAsyncClients are used to
-     * perform generic upload operations such as {@link BlockBlobAsyncClient#uploadFromFile(String) upload from file}
-     * and block blob specific operations such as {@link BlockBlobAsyncClient#stageBlockWithResponse(String, Flux, long,
-     * LeaseAccessConditions) stage block} and {@link BlockBlobAsyncClient#commitBlockList(List) commit block list},
-     * only use this when the blob is known to be a block blob.
-     *
-     * @return a {@link BlockBlobAsyncClient} created from the configurations in this builder.
-     * @throws NullPointerException If {@code endpoint}, {@code containerName}, or {@code blobName} is {@code null}.
-     */
-    public BlockBlobAsyncClient buildBlockBlobAsyncClient() {
-        return new BlockBlobAsyncClient(constructImpl(), snapshot, cpk);
-    }
+        Objects.requireNonNull(containerName);
```

Comment:
Add a message. Without it, you just get a NullPointerException with no message. ie. "'containerName' cannot be null." Same with another usages of this.

Reply:
Added message